.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateExtractor.cs

```diff
@@ -134,10 +134,20 @@ private List<Token> BasicRegexMatch(string text)
                 var matches = regex.Matches(text);
                 foreach (Match match in matches)
                 {
-                    // some match might be part of the date range entity, and might be splitted in a wrong way
+                    // some match might be part of the date range entity, and might be split in a wrong way
                     if (ValidateMatch(match, text))
                     {
-                        ret.Add(new Token(match.Index, match.Index + match.Length));
+                        var preText = text.Substring(0, match.Index);
+                        var relativeRegex = this.Config.RelativeRegex.MatchEnd(preText, trim: true);
+                        if (relativeRegex.Success)
+                        {
+                            ret.Add(new Token(relativeRegex.Index, match.Index + match.Length));

```

Comment:
You should check that the relative term is right before the detected entity no? Does match end already enforce it?
"I'm not talking about this, but about 4/Jan", for example. Please add it as a spec to show behaviour is correct.

Reply:
Yes, the matchEnd already enforce it, so i use the method.